### PR TITLE
Copter: make AutoTune object private state to ModeAutoTune

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Review by khancyr — APPROVED** [review]
LGTM, save 24bytes !

File: ArduCopter/mode.h [hunk 1]
```diff
@@ -548,6 +548,9 @@ class AutoTune : public AC_AutoTune
 
 class ModeAutoTune : public Mode {
 
+    // ParametersG2 sets a pointer within our autotune object:
+    friend class ParametersG2;
```
Comment (rmackay9) [inline ArduCopter/mode.h:552]: I thought the parameters were already available to modes using copter.g2.  This is what the other modes use so we might want to maintain that pattern.

Comment (peterbarker) [reply ArduCopter/mode.h:552]: This is the other way around - g2 has to modify a pointer within the private state of ModeAutoTune

**Review by rmackay9 — APPROVED** [review]

File: ArduCopter/mode.h (hunk 1 above)
Comment (peterbarker) [reply ArduCopter/mode.h:552]:
```
../../ArduCopter/Parameters.cpp: In constructor ‘ParametersG2::ParametersG2()’:
../../ArduCopter/Parameters.cpp:1087:41: error: ‘AutoTune ModeAutoTune::autotune’ is private within this context
     ,autotune_ptr(&copter.mode_autotune.autotune)
                                         ^~~~~~~~
compilation terminated due to -Wfatal-errors.

```

**Pull request merged**